src/Factory.sol

```diff
@@ -383,88 +391,90 @@ contract Factory {
         );
 
-        address distributor = DISTRIBUTOR_FACTORY.deploy(timelock, _vaultConfig.nodeOperatorManager);
-
-        address poolImpl = address(0);
-        if (poolType == STV_POOL_TYPE) {
-            poolImpl = STV_POOL_FACTORY.deploy(
-                dashboardAddress, _auxiliaryConfig.allowlistEnabled, withdrawalQueueProxy, distributor, poolType
-            );
-        } else if (poolType == STV_STETH_POOL_TYPE || poolType == STRATEGY_POOL_TYPE) {
-            poolImpl = STV_STETH_POOL_FACTORY.deploy(
-                dashboardAddress,
-                _auxiliaryConfig.allowlistEnabled,
-                _auxiliaryConfig.reserveRatioGapBP,
-                withdrawalQueueProxy,
-                distributor,
-                poolType
-            );
-        }
-
-        OssifiableProxy(payable(poolProxy))
-            .proxy__upgradeToAndCall(
-                poolImpl,
-                abi.encodeCall(StvPool.initialize, (tempAdmin, _commonPoolConfig.name, _commonPoolConfig.symbol))
-            );
-        OssifiableProxy(payable(poolProxy)).proxy__changeAdmin(timelock);
-
         intermediate = PoolIntermediate({
-            pool: poolProxy,
-            timelock: timelock,
-            strategyFactory: _strategyFactory,
-            strategyDeployBytes: _strategyDeployBytes
+            dashboard: dashboardAddress, poolProxy: poolProxy, withdrawalQueueProxy: wqProxy, timelock: timelock
         });
 
-        bytes32 deploymentHash = _hashIntermediate(intermediate, msg.sender);
+        bytes32 deploymentHash = _hashDeploymentConfiguration(
+            msg.sender,
+            _vaultConfig,
+            _commonPoolConfig,
+            _auxiliaryConfig,
+            _timelockConfig,
+            _strategyFactory,
+            _strategyDeployBytes,
+            intermediate
+        );
         uint256 finishDeadline = block.timestamp + DEPLOY_START_FINISH_SPAN_SECONDS;
         intermediateState[deploymentHash] = finishDeadline;
 
-        emit PoolCreationStarted(intermediate, finishDeadline);
+        emit PoolCreationStarted(
+            msg.sender,
+            _vaultConfig,
+            _commonPoolConfig,
+            _auxiliaryConfig,
+            _timelockConfig,
+            _strategyFactory,
+            _strategyDeployBytes,
+            intermediate,
+            finishDeadline
+        );
     }
 
     /// @notice Completes pool deployment (second phase)
+    /// @param _vaultConfig Configuration for the vault (must match createPoolStart)
+    /// @param _commonPoolConfig Common pool parameters (must match createPoolStart)
+    /// @param _auxiliaryConfig Additional pool configuration (must match createPoolStart)
+    /// @param _timelockConfig Configuration for the timelock controller (must match createPoolStart)
+    /// @param _strategyFactory Address of strategy factory (must match createPoolStart)
+    /// @param _strategyDeployBytes ABI-encoded parameters for strategy deployment (must match createPoolStart)
```

Comment:
consider using same natspec style as in other contract files. just for consistency